libnmstate/prettystate.py

```diff
@@ -56,12 +82,16 @@ def represent_ordereddict(dumper, data):
 
 
 def order_state(state):
-    iface_states = state.pop('interfaces', [])
-    state['interfaces'] = [
-        order_iface_state(iface_state) for iface_state in sorted(
-            iface_states, key=itemgetter('name')
-        )
-    ]
+    iface_states = state.pop('interfaces', None)
+
+    state = order_iface_state(state)
```

Comment:
I think this needs to be in the `else` block of the condition below.
If the input is "full", we either have an empty dict here or in the future something other than an interface state...
I would locate it below and add a comment about the assumption (assume that it is an iface state).

Reply:
When we add other things than interfaces, then the full input can contain both AFAIU. Therefore 

```
if interfaces:
...
elif routes:
...
else:
order_iface_state
```

will not work, because we need to order both interface and route states. Also he order_iface_state will also order all other dictionaries properly, therefore IMHO the current approach is good.